lib/model/util.go

```diff
@@ -59,3 +68,95 @@ func (d *deadlockDetector) Watch(name string, mut sync.Locker) {
 		}
 	}()
 }
+
+// moveforconflict renames a file to deal with sync conflicts
+func moveForConflict(name string, maxConflicts int) error {
```

Comment:
Okay I'll move the functions from util.go to folder.go as requested. Should hopefully have some time this weekend.